src/components/ScenarioParameters/components/ScenarioParametersTabs.js

```diff
@@ -30,23 +31,109 @@ const useStyles = makeStyles((theme) => ({
       color: theme.palette.primary.contrastText,
     },
+    '& .MuiTab-wrapper': {
+      '& .MuiSvgIcon-root': {
+        marginLeft: '5px',
+      },
+      flexDirection: 'row-reverse',
+    },
   },
   placeholder: {
     margin: `0 ${theme.spacing(3)}px`,
   },
 }));
 
-const ScenarioParametersTabs = ({ parametersGroupsMetadata }) => {
+function _buildScenarioTabList(tabs, userRoles, classes, t) {
+  const tabListComponent = [];
+  for (const groupMetadata of tabs) {
+    const lockedTab = !hasRequiredProfile(userRoles, groupMetadata.authorizedRoles);
+    if (lockedTab) {
+      if (groupMetadata.showParameterGroupIfNoPermission) {
+        tabListComponent.push(
+          <Tab
+            key={groupMetadata.id}
+            value={groupMetadata.id}
+            data-cy={groupMetadata.id + '_tab'}
+            icon={<LockIcon />}
+            className={classes.tab}
+            label={t(TranslationUtils.getParametersGroupTranslationKey(groupMetadata.id), groupMetadata.id)}
+          />
+        );
+      }
+    } else {
+      tabListComponent.push(
+        <Tab
+          key={groupMetadata.id}
+          value={groupMetadata.id}
+          data-cy={groupMetadata.id + '_tab'}
+          className={classes.tab}
+          label={t(TranslationUtils.getParametersGroupTranslationKey(groupMetadata.id), groupMetadata.id)}
+        />
+      );
+    }
+  }
+  return tabListComponent;
+}
+
+function _buildTabPanels(userRoles, tabs, classes) {
+  const tabPanelComponents = [];
+  for (let index = 0; index < tabs.length; index++) {
+    const groupMetadata = tabs[index];
+    const lockedTab = !hasRequiredProfile(userRoles, groupMetadata.authorizedRoles);
+    if (lockedTab) {
+      if (groupMetadata.showParameterGroupIfNoPermission) {
+        tabPanelComponents.push(
+          <TabPanel index={index} key={groupMetadata.id} value={groupMetadata.id} className={classes.tabPanel}>
+            {groupMetadata.tab}
+          </TabPanel>
+        );
+      }
+    } else {
+      tabPanelComponents.push(
+        <TabPanel index={index} key={groupMetadata.id} value={groupMetadata.id} className={classes.tabPanel}>
+          {groupMetadata.tab}
+        </TabPanel>
+      );
+    }
+  }
+  return tabPanelComponents;
+}
+
+const hasRequiredProfile = (userProfiles, requiredProfiles) => {
+  if (!requiredProfiles) {
+    return true;
+  }
+  if (Array.isArray(requiredProfiles) && requiredProfiles.length === 0) {
+    return true;
+  }
+  return requiredProfiles.some((profile) => userProfiles.includes(profile));
+};
+
+function chooseParametersTab(parametersGroupsMetadata, userRoles) {
+  const selectedTabId = '';
+  for (const groupMetadata of parametersGroupsMetadata) {
+    if (selectedTabId === '') {
+      const canViewTab = hasRequiredProfile(userRoles, groupMetadata.authorizedRoles);
+      if (canViewTab || groupMetadata.showParameterGroupIfNoPermission) {
+        return groupMetadata?.id;
+      }
+    }
+  }
+  return selectedTabId;
```

Comment:
Shouldn't we return `parametersGroupsMetadata?.[0]?.id` to handle the case where the user has access to none of the tabs ? If the function returns an empty string, the tab selector may not behave as expected.
IIUC the "forbidden" tab will be filled with a "You don't have access to this tab" placeholder anyway, right ?

Reply:
As discussion, it's good to leave it like that